src/controllers/netease/search.js

```diff
@@ -0,0 +1,86 @@
+// This module is intended to search songs
+const winston = require('winston')
+const sdk = require('NeteaseCloudMusicApi')
+const Joi = require('joi')
+const Cache = require('../../cache')
+const { md5 } = require('../../utils/crypto')
+// validation schema
+const { ValidateParams } = require('../../utils/response')
+const schema = Joi.object({
+  keyword: Joi.string().min(1).max(10000).required(),
+  limit: Joi.number().min(1).max(10000).default(30),
+  offset: Joi.number().min(0).default(0),
+  type: Joi.string().uppercase().valid(
+    'ALBUM',
+    'ARTIST',
+    'DJ',
+    'LYRIC',
+    'MV',
+    'PLAYLIST',
+    'SONG',
+    'USER',
+    'VIDEO',
+    'COMPLEX'
+  ).default('SONG'),
+  nocache: Joi.boolean().default(false)
+})
+
+async function getSearchDetail (params, ctx) {
```

Comment:
const { 
 keyword,
 limit,  
offset,
type,
} = params;

Reply:
ok, i will do it